drivers/flash/flash_stm32h7x.c

```diff
@@ -191,16 +264,20 @@ static int wait_write_queue(const struct device *dev, off_t offset)
 {
 	int64_t timeout_time = k_uptime_get() + 100;
 	FLASH_TypeDef *regs = FLASH_STM32_REGS(dev);
-	uint32_t sector = get_sector(offset);
+	struct flash_stm32_sector sector = get_sector(offset);
 	volatile uint32_t *sr;
 
-	if (sector < FLASH_SECTOR_TOTAL) {
+	switch (sector.bank) {
+	case 1:
 		sr = &regs->SR1;
+		break;
 #ifdef DUAL_BANK
```

Comment:
Same here. Put it in the `get_sector()` function.

Reply:
Done